### PR TITLE
DOC: Fix possessive "its" usage

### LINKED ISSUE #2997: Check and change typos `it's` -> `its`
The term `it's` can be found 171 times in ITK cxx, h, and hxx files. Check which of these instances should be changed to `its`. From https://github.com/InsightSoftwareConsortium/ITK/pull/2996#discussion_r775281256.

### PR DESCRIPTION
**Pull request by jamesobutler**
This closes #2997. I have reviewed and updated ITK files (not third-party sources) to update the correct usage of the possessive "Its". Note that I have not checked the inverse if any "its" usage should really be the contraction "it's", but this forgetting of an apostrophe for the contraction is seemingly not a common mistake.

Possessive "Its"
Correct: "Its class does something"

Contraction "It's" means "It is"
Correct: "It's correct to assume that something"

### CODE REVIEW

**Comment by seanm** [comment]
Nice!

But the third party changes, like to GDCM, should be dropped.

File: Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmCP246ExplicitDataElement.cxx [hunk 1]
```diff
@@ -26,7 +26,7 @@ VL CP246ExplicitDataElement::GetLength() const
     {
     assert( ValueField->GetLength().IsUndefined() );
     Value *p = ValueField;
-    // If this is a SQ we need to compute it's proper length
+    // If this is a SQ we need to compute its proper length
```
Comment (dzenanz) [inline Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmCP246ExplicitDataElement.cxx:29]: This is the only instance of changing third party code.

File: Documentation/ReleaseNotes/3.12.md [hunk 2]
```diff
@@ -1165,10 +1165,10 @@ Changes in this Release
         STYLE: according to kwstyle.
 
     Code/Common/itkImageBase
-        BUG: 8490: ImageBase::Initialized method modifies it's MTime requiring pipelines to re-executed. Added InitializeBufferedRegion method which does not call Modify as a protected method
+        BUG: 8490: ImageBase::Initialized method modifies its MTime requiring pipelines to re-executed. Added InitializeBufferedRegion method which does not call Modify as a protected method
```
Comment (dzenanz) [inline Documentation/ReleaseNotes/3.12.md:1168]: I am not sure whether we should be updating old release notes.

**Review by hjmjohnson — APPROVED** [review]
Nice.  Thanks for this change.

**Comment by jamesobutler** [comment]
> But the third party changes, like to GDCM, should be dropped.

Sorry, that one edit of a third party source snuck in there. I've updated the commit to remove it.

File: Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmCP246ExplicitDataElement.cxx (hunk 1 above)
Comment (hjmjohnson) [reply Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmCP246ExplicitDataElement.cxx]: @jamesobutler   Could you please revert the change to the GDCM file, `git commit --amend` and force push this PR again?

File: Documentation/ReleaseNotes/3.12.md (hunk 2 above)
Comment (hjmjohnson) [reply Documentation/ReleaseNotes/3.12.md:1168]: @dzenanz  I think we should update release notes for grammatical changes.

**Comment by jamesobutler** [comment]
Also if this is more of `STYLE` commit I can change it to that. It is modifying text in documentation files and also comments in other files.

**Comment by dzenanz** [comment]
I am fine with `DOC` prefix.

**Comment by hjmjohnson** [comment]
@jamesobutler This definitely seems like a DOC issue.

Incorrect grammar should not be a matter of preference.

**Pull request merged**